ntske.go

```diff
@@ -45,15 +44,15 @@ const (
 )
 
 func (s *Session) performKeyExchange() error {
-	tlsConfig := s.tlsConfig
-	if s.tlsConfig == nil {
-		tlsConfig = &tls.Config{}
+	dialer := s.options.Dialer
+	if dialer == nil {
+		dialer = func(ntskeAddress string, tlsConfig *tls.Config) (*tls.Conn, error) {
+			dialer := &net.Dialer{Timeout: s.options.Timeout}
+			return tls.DialWithDialer(dialer, "tcp", ntskeAddress, tlsConfig)
+		}
 	}
-	tlsConfig.MinVersion = tls.VersionTLS13
-	tlsConfig.NextProtos = []string{ntskeProtocol}
 
-	dialer := &net.Dialer{Timeout: time.Second * 5}
-	conn, err := tls.DialWithDialer(dialer, "tcp", s.ntskeAddr, tlsConfig)
+	conn, err := dialer(s.ntskeAddr, s.options.TLSConfig)
```

Comment:
nit: could we include the network as a parameter to be passed to the dialer? I know I could hard code `tcp` in the custom dialer, I just think it's more flexible to pass it as a param